playbooks/portals-deploy.yml

```diff
@@ -49,5 +49,8 @@
     portal_action: "portal-deploy"
 
   tasks:
-    - name: Including deploying portal
-      include_tasks: tasks/portal-deploy.yml
+    - name: Include deploying portal - start services
+      include_tasks: tasks/portal-deploy-start-services.yml
+
```

Comment:
My scenario is the following:
1. I'm a portal operator, running an open portal (no account required).
1. I decide to switch to authenticated-only.
1. I open `.env`, change the relevant config option and restart my portal.

At this point, my expectation would be that the portal will require authentication in order to use it and that all health checks and so on will function normally, i.e. pass. But that won't happen because there is no API key for the test user at that point.

Now my own thinking goes against my previous comment - there is no point in handling that during a deployment because it should be handled elsewhere - either in the tests themselves (somehow?) or via a portal reconfiguration action (`portal-setup-following`). So, instead of what I suggested above, maybe we should add a comment in the documentation and in `.env` that after changing the `ACCOUNTS_LIMIT_ACCESS` configuration option requires running `portal-setup-following`.

Reply:
If you want to switch to authenticated portal, you should configure authentication in LastPass (currently, maybe we can move this config to new: `ansible-private/custom-vars` as this is not secret, just a config) and you should rerun `portal-setup-following`, because there are more things needed to be set than just 1 env var, e.g. creating test user, because on anonymous portals we do not need, do not set it.

Reply:
> I'm a portal operator, running an open portal (no account required).
I decide to switch to authenticated-only.
I open .env, change the relevant config option and restart my portal.

Yea so IMO this scenario is the portal operator not following our recommended upgrade directions. 

As Fil stated, any config updates should require a re-run of the setup-following playbook.

Reply:
OK. If this is our preferred approach then feel free to resolve this.

I still feel that this extra step is somehow unnecessary and many people will forget it and it will lead to a lot of frustration.

Is it possible to add a start-up step that does the following:
1. Compare the current content of `.env` to a backed up copy created during the last successful run of `portal-setup-following` (ignoring empty lines and trailing whitespace).
2. If the content differs, automatically re-run `portal-setup-following`.

This would be a great QoL improvement for portal operators.

Reply:
I created an issue for this.

Portal operators should not be even aware, that there is an `.env` file and should never touch it manually.

All the configs are to be set in `hosts.ini` and customs yamls in ansible-private and sensitive configs in LastPass (or alternative secure storage(s) once they are implemented).